lib/ash_authentication/strategies/api_key/plug.ex

```diff
@@ -66,10 +66,12 @@ defmodule AshAuthentication.Strategy.ApiKey.Plug do
     required? = Keyword.get(opts, :required?, true)
     source = Keyword.get(opts, :source, :header)
     param_name = Keyword.get(opts, :param_name, "api_key")
-    header_prefix = Keyword.get(opts, :header_prefix, "Bearer ")
     subject_name = AshAuthentication.Info.authentication_subject_name!(resource)
     assign = Keyword.get(opts, :assign, :"current_#{subject_name}")
 
+    header_prefix = Keyword.get(opts, :header_prefix, "Bearer ")
+    :ok = validate_header_prefix(header_prefix)
```

Comment:
Last suggestion is not to match on the return here but just call it, and add a `!` to the function.